packages/database/src/modelClasses/AncestorDescendantRelation.js

```diff
@@ -41,22 +45,34 @@ export class AncestorDescendantRelationModel extends DatabaseModel {
   }
 
   async getChildIdToParentId(hierarchyId) {
-    const relationRecords = await this.getImmediateRelations(hierarchyId);
-    return reduceToDictionary(relationRecords, 'descendant_id', 'ancestor_id');
+    const cacheKey = this.getCacheKey(this.getChildIdToParentId.name, hierarchyId);
+    return this.runCachedFunction(cacheKey, async () => {
+      const relationRecords = await this.getImmediateRelations(hierarchyId);
+      return reduceToDictionary(relationRecords, 'descendant_id', 'ancestor_id');
+    });
```

Comment:
Decided to cache these individual functions rather than the base `getImmediateRelations()` function since that function returns a pretty large payload and caching it all would greatly increase the server memory footprint